Q-5. A certain city has two taxi companies: Company A has 80% of the taxis and
Company B has 20% of the taxis. Company A's taxis have a 95% success rate for picking
up passengers on time, while Company B's taxis have a 90% success rate. If a randomly
selected taxi is late, what is the probability that it belongs to Company A?

#### Ans:-
To calculate the probability that a randomly selected late taxi belongs to Company A, we can use Bayes' theorem. Bayes' theorem states that the probability of an event A given event B is equal to the probability of event B given event A multiplied by the probability of event A, divided by the probability of event B.

In [1]:
# Probability of selecting Company A's taxi
prob_company_a = 0.8

# Probability of selecting Company B's taxi
prob_company_b = 0.2

# Probability of a taxi from Company A being late
prob_late_given_a = 1 - 0.95  # 1 minus the success rate of Company A's taxis

# Probability of a taxi from Company B being late
prob_late_given_b = 1 - 0.9  # 1 minus the success rate of Company B's taxis

# Applying Bayes' theorem
prob_a_given_late = (prob_late_given_a * prob_company_a) / ((prob_late_given_a * prob_company_a) + (prob_late_given_b * prob_company_b))

print("The probability that a randomly selected late taxi belongs to Company A is:", prob_a_given_late)


The probability that a randomly selected late taxi belongs to Company A is: 0.6666666666666669


### The probability that a randomly selected late taxi belongs to Company A is: 0.6666666666666669
